src/Serilog/Capturing/PropertyValueConverter.cs

```diff
@@ -138,8 +138,18 @@ LogEventPropertyValue CreatePropertyValue(object? value, Destructuring destructu
             }
         }
 
-        if (value is string)
-            return new ScalarValue(value);
+        if (value is string s)
+            return s == string.Empty ? ScalarValue.EmptyString : new ScalarValue(value);
+
+        if (value is bool b)
+            return b ? ScalarValue.True : ScalarValue.False;
+
+        if (value is int i)
+        {
+            if (i == 0) return ScalarValue.Zero;
```

Comment:
We could use a switch expression here, that would eliminate two comparisons if `i == -1`.

Reply:
Switch expression should return value for each case.